api/v1alpha4/machinehealthcheck_types.go

```diff
@@ -56,6 +56,8 @@ type MachineHealthCheckSpec struct {
 
 	// Machines older than this duration without a node will be considered to have
 	// failed and will be remediated.
+	// If not set, this value is defaulted to 10 minutes.
```

Comment:
This should default to 0, nothing should be deleted without explicit user instructions.  Since this is an alpha level component, the time to change the API behavior is now and the impact is minimal.

Reply:
We shouldn't change the default behavior as part of this PR. That could break users workflow, if we're going to do that please open a different issue for v0.4 and the related PR should be marked as a breaking change.

Reply:
Most likely, nobody wants a 10m default.  I seriously doubt there's anyone that wants this on by default.  While it's remotely possible, it's probably not actually happening.

Let's not tie our hands on this on an alpha API with something that is obviously the wrong behavior.

Reply:
imo this is a tough one, i agree with both of you. i don't think we should be deleting machines unless a user is explicitly requesting it, but i also don't like breaking anyone who might have expected this behavior. not sure the best path forward, but i think i lean towards fixing this up to not delete the machines. just my 2 cents.

Reply:
I'm also not entirely sure on this one, but I think when we change the behaviour it's absolutely reasonable to ensure that it's documented as a breaking change.

Reply:
I created a new issue, https://github.com/kubernetes-sigs/cluster-api/issues/4497, please add any extra thoughts and continue the discussion there